src/network/gossip.rs

```diff
@@ -292,6 +302,31 @@ impl SnapchainGossip {
         })
     }
 
+    async fn get_announce_address(config: &Config) -> String {
+        if let Some(address) = &config.announce_address {
+            return address.clone();
+        }
+
+        // If no config-defined announce IP exists, detect the public IP.
+        // Falling back to the address also defined in the config
+        let public_ip = Self::get_public_ip().await;
+
+        match public_ip {
+            Ok(address) => format!("/ip4/{}/udp/{}/quic-v1", address, DEFAULT_GOSSIP_PORT),
+            Err(_) => config.address.clone(),
```

Comment:
log the error here

Reply:
Addressed! https://github.com/farcasterxyz/snapchain/pull/402/commits/5641a6c019b50324da2946c46a17a03ea1b9dcba